woocommerce/woocommerce-gateway-amazon-pay
# Fixed address street missing in billing address for German orders.

### All Submissions:


### Changes proposed in this Pull Request:

There is a special handling regarding the billing address for german orders that contemplates different scenarios.
However there is one missing that can lead to the billing address street to be empty.

If the buyer leaves `$address->AddressLine1` empty but fills `$address->AddressLine2`, no address will be saved. 

This is even how [Amazon in its documentation](https://amazonpaylegacyintegrationguide.s3.amazonaws.com/docs/eu/amazon-pay-automatic/sample-addresses-and-payment-instruments.html) passes the information. 

<img width="887" alt="image" src="https://user-images.githubusercontent.com/18492002/159477316-4c6084d9-93d0-406b-b64f-4812e70badb1.png">

Notice how the street number won't appear in the billing section.
![image](https://user-images.githubusercontent.com/18492002/159477275-c562524c-f06f-4ee0-9a12-c17fc38df4d1.png)


### How to test the changes in this Pull Request:

1. Create an Amazon Pay test account (Germany)
2. Make sure that address country code is DE
3. Create a billing address leaving the first field empty, but fill the second one.
4. Pay something.
5. Go to the the WooCommerce order page to check the billing address. 

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Fixed address street missing in billing address for German orders when $address->AddressLine1 is not present. 


Closes: https://github.com/woocommerce/woocommerce-gateway-amazon-pay/issues/203
